src/main/java/com/adzerk/sdk/Client.java

```diff
@@ -97,6 +117,14 @@ public DecisionResponse get(DecisionRequest request, final AdditionalOptions opt
             }
             if (opts.getIncludeExplanation() != null && opts.getIncludeExplanation()) {
               that.logger.info("Setting X-Adzerk-Explain.");
+              if (opts.getDesiredAds() != null) {
+                DesiredAdsObj desiredAdsObj= new DesiredAdsObj(opts.getApiKey(), opts.getDesiredAds());
+                builder.addHeader("X-Adzerk-Explain", desiredAdsObj.toString());
```

Comment:
This is going to use the object's `toString` method rather than converting to a JSON object. You'll have to use GSON to do the conversion to a JSON object.

Reply:
Updated